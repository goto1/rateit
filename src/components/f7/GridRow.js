import React from "react";
import PropTypes from "prop-types";

export const GridCol = ({ width, className, children }) => {
  const classNames = `col-${width ? width : "auto"} ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

GridCol.propTypes = {
  width: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

const GridRow = ({ noGutter = false, className, children }) => {
  const classNames = `row ${noGutter ? "no-gutter" : ""} ${className
    ? className
    : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

GridRow.propTypes = {
  noGutter: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default GridRow;
