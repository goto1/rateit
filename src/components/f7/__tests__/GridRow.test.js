import React from "react";
import GridRow, { GridCol } from "../GridRow";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("GridRow component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <GridRow>
          <GridCol>HelloWorld</GridCol>
          <GridCol>HelloWorld</GridCol>
        </GridRow>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with `no-gutter` styling class", () => {
    const wrapper = shallow(<GridRow noGutter={true}>HelloWorld</GridRow>);
    const actual = wrapper.props().className.includes("no-gutter");
    const expected = true;
    expect(actual).toBe(expected);
  });

  it("renders with a custom styling class", () => {
    const wrapper = shallow(<GridRow className="custom">HelloWorld</GridRow>);
    const actual = wrapper.props().className.includes("custom");
    const expected = true;
    expect(actual).toBe(expected);
  });

  describe("GridCol", () => {
    it("should render with an `auto` styling class when width not specified", () => {
      const wrapper = shallow(<GridCol>HelloWorld</GridCol>);
      const actual = wrapper.props().className.includes("col-auto");
      const expected = true;
      expect(actual).toBe(true);
    });

    it("should render with a specified width", () => {
      const wrapper = shallow(<GridCol width="50">HelloWorld</GridCol>);
      const actual = wrapper.props().className.includes("col-50");
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render with a custom styling class", () => {
      const wrapper = shallow(<GridCol className="custom">HelloWorld</GridCol>);
      const actual = wrapper.props().className.includes("custom");
      const expected = true;
      expect(actual).toBe(expected);
    });
  });
});
