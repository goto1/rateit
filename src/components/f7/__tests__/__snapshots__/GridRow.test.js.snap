// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow component should render correctly 1`] = `


<div
  className="row"
>
  <div
    className="col-auto"
  >
    HelloWorld
  </div>
  <div
    className="col-auto"
  >
    HelloWorld
  </div>
</div>
`;
